### PR TITLE
Const 4

### PR DESCRIPTION
**Pull request by nicolasbock**
This change is [Reviewable](https://reviewable.io/reviews/lanl/bml/273)

Depends-On: #271

### CODE REVIEW

File: src/C-interface/bml_add.h [hunk 1]
```diff
@@ -14,7 +14,7 @@ void bml_add(
 
 double bml_add_norm(
     bml_matrix_t * const A,
-    bml_matrix_t const * const B,
```
Comment (jeanlucf22) [inline src/C-interface/bml_add.h]: Shouldn't the first cont be in front of bml_matrix_t?

Comment (nicolasbock) [reply src/C-interface/bml_add.h]: According to section 6.7 of the C17 draft the order is not relevant and could be `const bml_matrix_t` or `bml_matrix_t const`. I can switch it around to read `const bml_matrix_t`. I agree, that this order somehow reads more naturally.

Comment (jmohdyusof) [reply src/C-interface/bml_add.h]: It would also be more readable if the * is attached (without a space) to one or other of the variables declarations.

Comment (nicolasbock) [reply src/C-interface/bml_add.h]: Yes, tru @jmohdyusof . I haven't figured out yet how to tell `indent` to do that. I'll keep trying.

Comment (jmohdyusof) [reply src/C-interface/bml_add.h]: Hi Nick, it looks like you got the const *const to work, but the other declarations still have 'floating' asterisks? Not sure how much you should worry about this.

Comment (nicolasbock) [reply src/C-interface/bml_add.h]: Yes, I played around with the command line arguments of `indent` but so far I haven't found a combination that places the `*` correctly in all cases. We can ignore it maybe for now and keep playing around with the arguments and then fix it if we find the right combination.

Comment (jmohdyusof) [reply src/C-interface/bml_add.h]: In my OMP offload versions I removed *all* the const declarations from the array pointers which were being written to (due to compiler errors), but this is probably overkill. I will see if the declarations you have allow compilation with the offload.

**Comment by nicolasbock** [comment]
I need a review to merge this though @jmohdyusof and @jeanlucf22

File: src/C-interface/bml_add.h (hunk 1 above)
Comment (nicolasbock) [reply src/C-interface/bml_add.h]: Perfect! Let me know how that goes! If this PR is doing the trick then we can go ahead and change the other uses of `const`.

**Comment by nicolasbock** [comment]
Thanks!

**Pull request merged**